src/addrman_impl.h

```diff
@@ -258,7 +265,7 @@ class AddrManImpl
      *
      *  @return  int The nid of the entry. If the addrman position is empty or not found, returns -1.
```

Comment:
nit:  The return type in the Doxygen content could be adjusted.
```diff
-     *  @return  int The nid of the entry. If the addrman position is empty or not found, returns -1.
+     *  @return  nid_type The nid of the entry. If the addrman position is empty or not found, returns -1.
      * */
     nid_type GetEntry(bool use_tried, size_t bucket, size_t position) const EXCLUSIVE_LOCKS_REQUIRED(cs);
```

nit / thinking out loud:  Returning `-1` is a remnant of the previous type being a plain `int`.  To keep with the theme of using the user-defined `nid_type`, would it be worth creating a macro (or constexpr, or similar) to name the invalid case (e.g. `INVALID_NID = -1`)?  Not sure, as I'm not seeing a need to adjust `nid_type` to anything more complicated than `int64_t`, and `GetEntry()` in this case is just indicating empty/not found.

Reply:
Fixed the return type.

Yes, also not sure about `INVALID_NID` - I don't really see why we would want to change it from -1 to another value, but happy to add it if others like it... I'll keep this thread open for a while.